sqlgen/db.go

```diff
@@ -123,6 +140,29 @@ func (db *DB) execWithTrace(ctx context.Context, query SQLQuery, operationName s
 	return db.QueryExecer(ctx).ExecContext(ctx, clause, args...)
 }
 
+// Count counts the number of relevant rows in a database, matching options in filter
+//
+// model should be a pointer to a struct, for example:
+//
+//   var res int64
+//   user := &User{Id; 10}
+//   if err := db.Count(ctx, user, &res, Filter{}); err != nil {
+//
+func (db *DB) Count(ctx context.Context, model interface{}, result *int64, filter Filter) error {
```

Comment:
Drive by: I think there's no reason to use the pattern of a result pointer argument since the return type is fixed.  I'd return (int64, error).  It's a less awkward API.

Reply:
totally agreed, I want to get some opinions from other maintainers, as all of the other API's return singleton error

Reply:
@jellevandenhooff similarly prefers the less clunky direct return!